### PR TITLE
Hlint redundancies.

### PR DESCRIPTION
**Pull request by philderbeast**
See #1474 for redundant id, redundant return, redundant if, redundant <$>, redundant map and redundant lambda.

### CODE REVIEW

**Comment by Kleidukos** [comment]
@mergifyIO rebase

**Comment by philderbeast** [comment]
@Kleidukos I rebased, got HLint running again (it was barfing on InterfaceFile, https://github.com/ndmitchell/hlint/issues/1442) and followed hints about rendundancies except for redundant brackets. They're numerous and I'll do those on a separate pull request.

File: haddock-test/src/Test/Haddock/Config.hs [hunk 1]
```diff
@@ -220,7 +220,7 @@ loadConfig ccfg dcfg flags files = do
 
     cfgDiffTool <- if FlagNoDiff `elem` flags
         then pure Nothing
-        else (<|>) <$> pure (flagsDiffTool flags) <*> defaultDiffTool
+        else pure ((flagsDiffTool flags) <|>) <*> defaultDiffTool
```
Comment (Kleidukos) [inline haddock-test/src/Test/Haddock/Config.hs:223]: I'm not sure this brings any value, the original line is clearer

**Review by Kleidukos — DISMISSED** [review]

**Review by Kleidukos — CHANGES_REQUESTED** [review]
See comments. :) We're nearly there!

**Comment by philderbeast** [comment]
Thanks for the review @Kleidukos. Closing here. I may create a new merge request for it over in the GHC repo.

**Pull request closed without merging**